tests/src/packages/websocket/WebSocketTests.scala

```diff
@@ -0,0 +1,116 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package packages.websocket
+
+import java.net.URI
+
+import org.junit.runner.RunWith
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.Finders
+import org.scalatest.junit.JUnitRunner
+
+import common.TestHelpers
+import common.Wsk
+import common.WskProps
+import common.WskTestHelpers
+import spray.json.DefaultJsonProtocol.StringJsonFormat
+import spray.json.pimpAny
+
+@RunWith(classOf[JUnitRunner])
+class WebSocketTests
+        extends TestHelpers
+        with WskTestHelpers
+        with BeforeAndAfterAll {
+
+    implicit val wskprops = WskProps()
+    val wsk = new Wsk()
+    var serverURI: URI = WebSocketUtils.startWebSocketServer()
+
+    val websocketSendAction = "/whisk.system/websocket/send"
+
+    behavior of "Websocket action"
+
+    override def afterAll() {
+        WebSocketUtils.stopWebSocketServer()
+    }
+
+    it should "Stand up a websocket server for testing" in {
+        println(s"Server URI: ${serverURI}")
+
+        var serverStarted = false
+        var retryAttempt = 0
+
+        // The test server starts/runs in the background. Let's give it some time
+        // to finish initializing before declaring that it isn't running.
+        while (!serverStarted && retryAttempt < 3) {
+            retryAttempt = retryAttempt + 1
+
+            println(s"Contacting test websocket server attempt #${retryAttempt}")
+            val testClient = WebSocketUtils.getSocketClient(serverURI)
+            serverStarted = testClient.connectBlocking()
+
+            if (!serverStarted) {
+                Thread.sleep(1000)
+            } else {
+                testClient.closeBlocking()
+            }
+        }
+
+        serverStarted should be(true)
+    }
+
+    it should "Use the websocket action to send a payload" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val run = wsk.action.invoke(websocketSendAction, Map("uri" -> serverURI.toString.toJson, "payload" -> "This is the message to send".toJson))
+            withActivation(wsk.activation, run) {
+                activation =>
+                    activation.fields("response").asJsObject.fields("success").toString should be("true")
+            }
+    }
+
+    it should "Return an error due to a malformed URI" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val badURI = new URI("ws://localhost:80")
+
+            val run = wsk.action.invoke(websocketSendAction, Map("uri" -> badURI.toString.toJson, "payload" -> "This is the message to send".toJson))
+            withActivation(wsk.activation, run) {
+                activation =>
+                    activation.fields("response").asJsObject.fields("success").toString should be("false")
+                    activation.fields("response").asJsObject.fields("result").asJsObject.fields("error") should not be (null)
```

Comment:
`fields` documentation states:

> Retrieves the value which is associated with the given key. This method invokes the default method of the map if there is no mapping from the given key to a value. Unless overridden, the default method throws a NoSuchElementException.

if you want to check that field doesn't exist, you can use `getFields` and check the sequence instead.